docs/source/user_guides/optimization_guide.rst

```diff
@@ -0,0 +1,158 @@
+the available options including
+* JSON: Uses JSON files for storage.
+* Pickle: Utilizes Python's Pickle format for storage.
+* Shelve: Utilizes Python's Shelve format for storage.
+* PostgreSQL: A PostgreSQL database for storage.
+* MongoDB: A MongoDB database for storage.
+* Redis: A Redis database for storage.
+* MySQL: A MySQL database for storage.
+* SQLite: An SQLite database for storage.
+* YDB: Utilizes YDB (Yandex Database) for storage.
+
+For some storage methods like Pickle, Shelve, and SQLite, you may need to create temporary directories or files
+to perform the benchmarking.
+
+**Saving Benchmark Results to a File**
+
+Benchmark results are saved to files for analysis and comparison.
+There are two functions available for this purpose:
+
+* benchmark_all: This function is a higher-level wrapper that accepts benchmark cases and configurations for multiple databases in a single file.
+* save_results_to_file: This function accepts a single URI for the database and multiple benchmark configurations.
+
+Both functions use a `BenchmarkConfig` to configure benchmark behavior.
+The `BenchmarkConfig` allows you to set parameters
+like the number of contexts that will be saved and retrieved from the storage during testing (context_num)
+and the dimensions of messages and other components,
+allowing for fine-grained control over the benchmarks.
+
+**File Structure**
+
+Benchmark results are saved according to a specific schema,
+which can be found in the benchmark schema documentation.
+Each database being benchmarked will have its own result file.
+
+**Viewing Benchmark Results**
+
+Once the benchmark results are saved to a file, you can view and analyze them using two methods:
+
+* Using the Report Function: The report function can display specified information from a given file. By default, it prints the name and average metrics for each benchmark case.
+* Using the Streamlit App: A Streamlit app is available for viewing and comparing benchmark results. You can upload benchmark result files using the app's "Benchmark sets" tab, inspect individual results in the "View" tab, and compare metrics in the "Compare" tab.
+
+**Additional Information**
+
+DFF provides configuration presets in the `basic config <../apiref/dff.utils.db_benchmarks.basic_config.py>`_ module,
+covering various contexts, messages, and edge cases.
+You can use these presets by passing them to the benchmark functions.
+
+If the basic configurations are not sufficient for your needs, you can create custom configurations by inheriting from the BenchmarkConfig class.
+You'll need to define three methods:
+
+* `get_context` for getting initial contexts,
+* `info` for getting display information representing the configuration,
+* and `context_updater` for updating contexts.
+
+.. note::
+
+    The exact instructions of how the testing can be carried out are available in the
+    `DB benchmarking tutorial <../tutorials/tutorials.context_storages.8_db_benchmarking.py>`_.
```

Comment:
I think that instead of mentioning all the information above in this guide it would be better to move it to the db benchmarking tutorial and simply link it here.

Reply:
Updated